Userland/Applications/TextEditor/MainWidget.cpp

```diff
@@ -272,28 +275,36 @@ MainWidget::MainWidget()
                 return;
         }
 
-        open_file(open_path.value());
+        open_file(fd_response.fd()->take_fd(), fd_response.chosen_file().value());
     });
 
     m_save_as_action = GUI::CommonActions::make_save_as_action([&](auto&) {
-        Optional<String> save_path = GUI::FilePicker::get_save_filepath(window(), m_name.is_null() ? "Untitled" : m_name, m_extension.is_null() ? "txt" : m_extension);
-        if (!save_path.has_value())
+        auto response = m_filesystem_client->prompt_save_file(m_name.is_null() ? "Untitled" : m_name, m_extension.is_null() ? "txt" : m_extension, Core::StandardPaths::home_directory(), O_WRONLY | O_CREAT | O_TRUNC);
+
+        if (response.error() != 0)
             return;
 
-        if (!m_editor->write_to_file(save_path.value())) {
+        if (!m_editor->write_to_file_and_close(response.fd()->take_fd())) {
             GUI::MessageBox::show(window(), "Unable to save file.\n", "Error", GUI::MessageBox::Type::Error);
             return;
         }
         // FIXME: It would be cool if this would propagate from GUI::TextDocument somehow.
         window()->set_modified(false);
 
-        set_path(LexicalPath(save_path.value()));
-        dbgln("Wrote document to {}", save_path.value());
+        set_path(LexicalPath(response.chosen_file().value()));
+        dbgln("Wrote document to {}", response.chosen_file().value());
```

Comment:
Just as an advance warning, this will become a merge conflict as soon as #8320 is merged.